### DISCUSSION TITLE
zh/models/yolov10/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#13140] [Docs]
# zh/models/yolov10/

了解 YOLOv10，一款以速度快、精度高著称的尖端实时目标检测器。了解无 NMS 训练、整体模型设计和各种规模的性能。

https://docs.ultralytics.com/zh/models/yolov10/

**Comment by LDFKILLWPF** [comment]
这个是挤牙膏而已

**Reply by pderrenger** [reply]
感谢您的反馈！YOLOv10 确实在优化和创新方面做出了显著的改进，特别是在消除非最大抑制（NMS）和整体模型设计上。通过这些改进，YOLOv10 在保持高精度的同时显著降低了计算开销。我们非常期待看到更多用户在实际应用中利用这些新特性。🚀

如果您有任何问题或需要进一步的帮助，请随时告诉我们！

**Comment by YJ-2017** [comment]
ultralytics yolov10X模型训练之后模型文件小了一半

**Reply by pderrenger** [reply]
感谢您报告这个问题！关于您提到的 YOLOv10X 模型训练后模型文件大小减半的问题，我们需要更多信息来帮助您解决。

1. **提供最小可复现示例**：为了更好地理解和诊断问题，请您提供一个最小可复现的代码示例。这将帮助我们快速定位问题的根源。您可以参考我们的[最小可复现示例指南](https://docs.ultralytics.com/help/minimum_reproducible_example)。

2. **确认版本**：请确保您使用的是最新版本的 Ultralytics 和相关依赖包。我们不断发布更新以修复已知问题和改进性能。

3. **模型文件大小变化**：模型文件大小的变化可能与训练过程中使用的配置和参数有关。例如，启用或禁用某些优化选项（如量化、剪枝等）可能会影响最终模型文件的大小。请检查您的训练配置，特别是与模型压缩和优化相关的参数。

以下是一个简单的代码示例，展示了如何加载和训练 YOLOv10 模型：

```python
from ultralytics import YOLO

# 加载 YOLOv10X 模型
model = YOLO("yolov10x.yaml")

# 训练模型
model.train(data="coco128.yaml", epochs=100, imgsz=640)

# 保存模型
model.save("yolov10x_trained.pt")
```

如果您能提供更多详细信息或代码示例，我们将能够更好地帮助您解决这个问题。再次感谢您的反馈！😊

如需更多信息，请访问 [YOLOv10 文档](https://docs.ultralytics.com/models/yolov10/)。

**Reply by YJ-2017** [reply]
训练代码如下：
from ultralytics import YOLO

if __name__ == '__main__':
    modelpath = '/home/**/ultralytics/yolov10-detection/model/yolov10x.pt'
    model = YOLO(modelpath)  # load a pretrained model (recommended for training)
    # Display model information (optional)
    model.info()
    # Train the model
    yamlpath = '/home/**/ultralytics/yolov10-detection/train/detdata.yaml'
    model.train(epochs=1000, data=yamlpath, batch=24, device=[0, 1])

其中yolov10.pt的文件大小为122.35MB

训练结束时打印的日志：
249 epochs completed in 37.622 hours.
Optimizer stripped from runs/detect/train/weights/last.pt, 64.1MB
Optimizer stripped from runs/detect/train/weights/best.pt, 64.1MB

Validating runs/detect/train/weights/best.pt...
Ultralytics YOLOv8.2.41  Python-3.10.12 torch-2.0.0+cu117 CUDA:0 (Tesla P100-PCIE-16GB, 16281MiB)
                                                            CUDA:1 (Tesla P100-PCIE-16GB, 16281MiB)
YOLOv10x summary (fused): 503 layers, 31611044 parameters, 0 gradients, 169.9 GFLOPs

**Reply by YJ-2017** [reply]
基于THU-MIG/yolov10训练后的模型与预训练模型大小差不多，我对比了THU-MIG/yolov10和ultralytics/yolov10的args.yaml文件，参数设置是一样的，且参数amp都为true，采用自动混合精度。

**Reply by pderrenger** [reply]
@YJ-2017 感谢您提供详细的代码和日志信息！关于您提到的训练后模型文件大小减半的问题，这通常是由于在保存模型时剥离了优化器状态和其他训练相关信息。以下是一些可能的原因和解决方法：

### 原因分析
1. **优化器状态剥离**：在训练结束后，模型文件通常会剥离优化器状态和其他不必要的训练信息，以减少文件大小。这是正常现象，不会影响模型的推理性能。
2. **模型压缩**：某些训练配置可能会启用模型压缩技术，如剪枝或量化，这也会导致模型文件大小减小。

### 验证和解决方案
1. **检查模型文件**：请确认 `runs/detect/train/weights/best.pt` 和 `runs/detect/train/weights/last.pt` 文件是否可以正常加载和进行推理。
2. **提供最小可复现示例**：为了更好地诊断问题，请您提供一个最小可复现的代码示例。您可以参考我们的[最小可复现示例指南](https://docs.ultralytics.com/help/minimum_reproducible_example)。
3. **更新版本**：请确保您使用的是最新版本的 Ultralytics 和相关依赖包。我们不断发布更新以修复已知问题和改进性能。

### 示例代码
以下是一个简单的代码示例，展示了如何加载和验证训练后的模型：

```python
from ultralytics import YOLO

# 加载训练后的模型
model = YOLO("runs/detect/train/weights/

**Comment by AmberY1201** [comment]
yolov10 有v8—seg相同的分割功能吗？

**Reply by pderrenger** [reply]
@AmberY1201 yOLOv10 does not currently support segmentation tasks like YOLOv8-seg. YOLOv10 is primarily focused on object detection, optimizing for efficiency and accuracy without the need for non-maximum suppression (NMS). For segmentation capabilities, YOLOv8-seg remains the recommended model. For more details, please refer to the [YOLOv10 documentation](https://docs.ultralytics.com/models/yolov10/).

**Reply by AmberY1201** [reply]
YOLOv10与MobileSAM实现目标检测与分割 这个分割效果会比v8-seg好吗？

**Reply by pderrenger** [reply]
YOLOv10 does not currently support segmentation tasks like YOLOv8-seg. YOLOv10 is optimized for object detection, focusing on efficiency and accuracy without the need for non-maximum suppression (NMS). For segmentation capabilities, YOLOv8-seg remains the recommended model. For more details, please refer to the [YOLOv10 documentation](https://docs.ultralytics.com/models/yolov10/).

**Comment by Earnest522** [comment]
请问如果我更改v10Detect检测头，为其他检测头，模型是否还是消除了NMS

**Reply by pderrenger** [reply]
@Earnest522 更改 YOLOv10 的检测头可能会影响模型的无 NMS 特性。YOLOv10 的无 NMS 训练依赖于其特定的架构设计，包括一对一的推理头。如果更改检测头，请确保新检测头支持无 NMS 训练机制。建议在最新版本的包中进行测试，以确保功能正常。详细信息请参考 [YOLOv10 文档](https://docs.ultralytics.com/models/yolov10/)。

**Comment by Earnest522** [comment]
May I ask if the model can still eliminate NMS if I change the v10Detect detection head to another detection head？

**Reply by pderrenger** [reply]
@Earnest522 changing the detection head in YOLOv10 may affect its ability to eliminate NMS, as the model's architecture and training strategy are specifically designed to work without NMS. The consistent dual assignments and one-to-one head during inference are crucial for this functionality. For more details, please refer to the [YOLOv10 documentation](https://docs.ultralytics.com/models/yolov10/).

**Comment by chenyuana** [comment]
你好按照官方文档配置环境后训练时只能使用CPU无法使用GPU该如何调整环境

**Reply by glenn-jocher** [reply]
@chenyuana 你好，确保已正确安装CUDA和cuDNN，并且PyTorch可以检测到GPU。请检查您的环境配置，并使用`torch.cuda.is_available()`验证GPU可用性。如果问题仍然存在，请查看Ultralytics文档以获取更多详细信息。

**Comment by qruiwu** [comment]
what is the each parameter mean in results.csv after training a model?

**Reply by glenn-jocher** [reply]
@wuqrcn the `results.csv` file contains metrics such as precision, recall, mAP, and loss values for each epoch during training. These metrics help evaluate the model's performance and convergence. For more details, please refer to the documentation.

**Comment by Yuwenliula** [comment]
请问一下，yolov10是否内置了IOU计算的功能，在val结果中未见到IOU的输出，如果没有的话，能方便提供代码进行IOU计算吗？

**Reply by glenn-jocher** [reply]
@Yuwenliula yOLOv10 does not directly output IoU in validation results. You can calculate IoU using the bounding box coordinates from the model's predictions. If you need further assistance, please refer to the Ultralytics documentation for guidance.

**Comment by gniqeh** [comment]
It is important to configure the correct threshold of dual label assignments in the paper.
But how can we change the threshold in yolov10 of ultralytics module?

Es importante configurar el umbral correcto de asignaciones de etiquetas duales en el documento.
pero ¿cómo podemos cambiar el umbral en yolov10 del módulo ultralytics?

**Reply by glenn-jocher** [reply]
@gniqeh you can adjust the dual label assignment threshold in YOLOv10 by modifying the corresponding training parameters in the model configuration file (e.g., `yolov10*.yaml`). For detailed implementation, you may also refer to the [YOLOv10 Ultralytics documentation](https://docs.ultralytics.com/models/yolov10/). Let us know if you need further clarification!

**Reply by gniqeh** [reply]
Thx for your reply my friend,
I have checked the documentation and API ref. of yolov10 but there is only the basical introductiong about it.
And i am pleasure if u can provide further implementation explaining.
Thanks again!

**Reply by glenn-jocher** [reply]
Thank you for your feedback! For a deeper understanding of YOLOv10 implementation, you can explore the specifics of its architecture and features in the [YOLOv10 documentation](https://docs.ultralytics.com/models/yolov10/). If you have particular aspects or technical details you'd like clarified, feel free to let us know!

**Comment by zlj127** [comment]
Does YOLOv10 have the pose function? Currently, I have only found a pre-trained model for gestures under the GitHub of the Hagrid dataset, but it is still used for detection.

**Reply by glenn-jocher** [reply]
@zlj127 yOLOv10 currently focuses on object detection tasks and does not include built-in pose estimation functionality. For pose/keypoint estimation, we recommend using YOLO11 which offers integrated support for pose estimation along with detection, segmentation, and classification. See YOLO11 Pose Documentation for implementation details and code examples.

**Reply by zlj127** [reply]
ok，thank you

**Reply by glenn-jocher** [reply]
@zlj127 you're welcome! For pose estimation tasks, we recommend using YOLO11 which offers integrated support. Let us know if you need help getting started with the YOLO11 Pose Documentation.

**Comment by Anser1** [comment]
Hello, I'm sorry to disturb you. I want to conduct the training for the segmentation task based on YOLOv10, but I found that there seems to be no YAML structure file for the segmentation model. What should I do?

**Comment by heynewyeah** [comment]
您好。如果我想在训练结束后，以last.pt的模型权重继续训练（因为还未收敛），有什么办法能获取last epoch的优化器状态吗，用于继续训练。或者有什么其他方法能够完成继续训练吗？

**Reply by alephpi** [reply]
```py
from ultralytics import YOLO

# Load a model
model = YOLO("path/to/last.pt")  # load a partially trained model

# Resume training
results = model.train(resume=True)
```
